src/test/ui/transmutability/enums/should_order_correctly.rs

```diff
@@ -0,0 +1,32 @@
+// check-pass
+//! The payloads of an enum variant should be ordered after its tag.
+
+#![crate_type = "lib"]
+#![feature(arbitrary_enum_discriminant)]
+#![feature(transmutability)]
+#![allow(dead_code)]
+
+mod assert {
+    use std::mem::BikeshedIntrinsicFrom;
+    pub struct Context;
+
+    pub fn is_transmutable<Src, Dst>()
+    where
+        Dst: BikeshedIntrinsicFrom<Src, Context, true, true, true, true>
+    {}
+}
+
+#[repr(u8)] enum V0 { V = 0 }
+#[repr(u8)] enum V1 { V = 1 }
+#[repr(u8)] enum V2 { V = 2 }
+
+#[repr(u8)] enum E01 { V0(V1) = 0u8 }
+#[repr(u8)] enum E012 { V0(V1, V2) = 0u8 }
```

Comment:
uh... this seems fishy...

```
[src/main.rs:11] std::mem::size_of::<E012>() = 3
```

or is that transmute a truncating transmute? (if so, maybe document all of the 

```rust
BikeshedIntrinsicFrom<Src, Context, X, Y, Z, A>
```

Reply:
Yeah, it's a truncating transmute. None of the safety options ("X, Y, Z, A") correspond to allowing/disallowing truncation, because truncation is irrelevant to safety. (The intuition for this is that transmutes are basically `memcpy`s, and leaving off bytes at the end of the copied range cannot affect the validity of bytes at the beginning of the range).